src/linear_algebra/finsupp.lean

```diff
@@ -141,247 +101,252 @@ begin
     refine subset.trans (finset.coe_subset.2 support_smul) hp }
 end
 
-lemma mem_restrict_dom {s : set α} (p : α →₀ β) : p ∈ (restrict_dom β γ s) ↔ ↑p.support ⊆ s :=
-iff.rfl
+variables {β}
 
-section
-set_option class.instance_max_depth 37
-def restrict_dom_equiv_finsupp (s : set α) [decidable_pred (λ (x : α), x ∈ s)] :
-  (restrict_dom β γ s) ≃ₗ[γ] (s →₀ β) :=
-(restrict_support_equiv s).to_linear_equiv
-begin
-  show is_linear_map γ ((lsubtype_domain s : (α →₀ β) →ₗ[γ] (s →₀ β)).comp
-      (submodule.subtype (restrict_dom β γ s))),
-  exact linear_map.is_linear _
-end
-end
+lemma mem_supported {s : set α} (p : α →₀ β) : p ∈ (supported β γ s) ↔ ↑p.support ⊆ s :=
+iff.rfl
 
-end
+lemma mem_supported' {s : set α} (p : α →₀ β) :
+  p ∈ supported β γ s ↔ ∀ x ∉ s, p x = 0 :=
+by simp [mem_supported, set.subset_def, not_imp_comm]
 
-end module
+lemma single_mem_supported {s : set α} {a : α} (b : β) (h : a ∈ s) :
+  single a b ∈ supported β γ s :=
+set.subset.trans support_single_subset (set.singleton_subset_iff.2 h)
 
-section vector_space
-variables {α : Type*} {β : Type*} {γ : Type*}
-variables [decidable_eq α] [decidable_eq β] [discrete_field γ] [add_comm_group β] [vector_space γ β]
+lemma supported_eq_span_single [has_one β] (s : set α) :
+  supported γ γ s = span γ ((λ i, single i 1) '' s) :=
+begin
+  refine (span_eq_of_le _ _ (le_def'.2 $ λ l hl, _)).symm,
+  { rintro _ ⟨_, hp, rfl ⟩ , exact single_mem_supported γ 1 hp },
+  { rw ← l.sum_single,
+    refine sum_mem _ (λ i il, _),
+    convert @smul_mem γ (α →₀ γ) _ _ _ _ (single i 1) (l i) _,
+    { simp },
+    apply subset_span,
+    apply set.mem_image_of_mem _ (hl il) }
+end
 
-open linear_map submodule
+variables (β γ)
 
-instance : vector_space γ (α →₀ β) :=
-{ to_module := finsupp.module }
+def restrict_dom (s : set α) : α →₀ β →ₗ supported β γ s :=
```

Comment:
The parentheses here seem weird. I suggest making all the arrows be right assoc with the same precedence, and put parens here so it's not misleading.

Reply:
Done.